File: LeetCode (en comments)/125. Valid Palindrome.go
func isPalindrome(s string) bool {
    var cleaned strings.Builder
    for _, char := range strings.ToLower(s) {
        if unicode.IsLetter(char) || unicode.IsDigit(char) {
            cleaned.WriteRune(char)
        }
    }
    cleanedStr := cleaned.String()
    l := len(cleanedStr)
    for i := 0; i < l / 2; i++ {
        if cleanedStr[i] != cleanedStr[l - 1 - i] {
            return false
        }
    }
    return true
}

File: LeetCode (en comments)/125. Valid Palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = [c.lower() for c in s if c.isalnum()]
        l = len(s)
        for i in range(len(s) // 2):
            if s[i] != s[l - 1 - i]:
                return False
        return True


class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = [c.lower() for c in s if c.isalnum()]
        return s == s[::-1]